### PR TITLE
Allow transformed columns in JoinGroupby

### PR DESCRIPTION
**Pull request by benfred**
The JoinGroupby op previously could only handle calculating stats on 'cont_cols'
that were unchanged from the original dataset. This change makes it so that
you can use arbitrary columns - including ones that have had some transformations
applied already.

### CODE REVIEW

**Comment by benfred** [comment]
rerun tests

File: nvtabular/ops/join_groupby.py [hunk 1]
```diff
@@ -39,14 +40,14 @@ class JoinGroupby(StatOperator):
...
         # Use JoinGroupby to define a NVTabular workflow
         groupby_features = ['cat1', 'cat2', 'cat3'] >> ops.JoinGroupby(
-            out_path=str(tmpdir), stats=['sum','count'], cont_names=['num1']
+            out_path=str(tmpdir), stats=['sum','count'], cont_cols=['num1']
         )
         processor = nvtabular.Workflow(groupby_features)
 
     Parameters
     -----------
-    cont_names : list of str
```
Comment (bschifferer) [inline nvtabular/ops/join_groupby.py:48]: Why do we rename cont_names to cont_cols?

**Review by bschifferer — COMMENTED** [review]
Do we need a unit test to capture the edge case?

File: nvtabular/ops/join_groupby.py (hunk 1 above)
Comment (benfred) [reply nvtabular/ops/join_groupby.py:48]: I'm expanding what it does here - its not just the column names anymore, but can alos be a columngroup or transformed columns now.

**Comment by benfred** [comment]
> Do we need a unit test to capture the edge case?

I've added a unittest as part of this PR (using JoinGroupBy with a normalized continuous column) in [test_joingroupby_dependency](https://github.com/NVIDIA/NVTabular/pull/670/files#diff-e4d18001a1d407273320606db51ffe8ef20709522220cdc6e3cde2070f06b246R696). Is there another test case you

**Comment by bschifferer** [comment]
> > Do we need a unit test to capture the edge case?
> 
> I've added a unittest as part of this PR (using JoinGroupBy with a normalized continuous column) in [test_joingroupby_dependency](https://github.com/NVIDIA/NVTabular/pull/670/files#diff-e4d18001a1d407273320606db51ffe8ef20709522220cdc6e3cde2070f06b246R696). Is there another test case you have in mind?

Ok - sorry didnt see that one :)

**Review by bschifferer — APPROVED** [review]
